jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -43,33 +45,47 @@ object EcsExecUtils : CoroutineScope by ApplicationThreadPoolScope("EcsExec") {
         }
     }
 
-    suspend fun checkServiceState(project: Project, service: Service, enable: Boolean) {
-        /* There appears to be a lag between the time the UpdateService call is made and
-         the time the changes are surfaced via the DescribeServices call.
-         Hence a delay of 5000 milliseconds is added before polling for the completion state of the service
-         */
-        delay(5000)
-        val request = DescribeServicesRequest.builder().cluster(service.clusterArn()).services(service.serviceArn()).build()
-        val client = project.awsClient<EcsClient>()
-        val waiter = client.waiter()
-        try {
-            waiter.waitUntilServicesStable(request)
-            project.refreshAwsTree(EcsResources.describeService(service.clusterArn(), service.serviceArn()))
-            if (enable) {
-                notifyInfo(message("ecs.execute_command_enable"), message("ecs.execute_command_enable_success", service.serviceName()))
-                EcsTelemetry.enableExecuteCommand(project, Result.Succeeded)
-            } else {
-                notifyInfo(message("ecs.execute_command_disable"), message("ecs.execute_command_disable_success", service.serviceName()))
-                EcsTelemetry.disableExecuteCommand(project, Result.Succeeded)
-            }
-        } catch (e: Exception) {
-            if (enable) {
-                notifyError(message("ecs.execute_command_enable"), message("ecs.execute_command_enable_failed", service.serviceName()))
-                EcsTelemetry.enableExecuteCommand(project, Result.Failed)
-            } else {
-                notifyError(message("ecs.execute_command_disable"), message("ecs.execute_command_disable_failed", service.serviceName()))
-                EcsTelemetry.disableExecuteCommand(project, Result.Failed)
-            }
+    private fun checkServiceState(project: Project, service: Service, enable: Boolean) {
+        val title = if (enable) {
+            message("ecs.execute_command_enable_progress_indicator_message")
+        } else {
+            message("ecs.execute_command_disable_progress_indicator_message")
         }
+        ProgressManager.getInstance().run(
+            object : Task.Backgroundable(project, title, false) {
+                override fun run(indicator: ProgressIndicator) {
+                    /* There appears to be a lag between the time the UpdateService call is made and
+                    the time the changes are surfaced via the DescribeServices call.
+                    Hence a delay of 5000 milliseconds is added before polling for the completion state of the service
+                    */
+                    Thread.sleep(5000)
+                    val request = DescribeServicesRequest.builder().cluster(service.clusterArn()).services(service.serviceArn()).build()
+                    val client = project.awsClient<EcsClient>()
```

Comment:
What happens if the customer changes the profile/region in the 5 seconds between when they call the action and this starts?

I think we probably actually want to pass the connection along, rather than assuming that the "current connection" is going to remain static.

Reply:
Currently, if they change the region/credentials within the 5 secs they will get a notification saying the command could not be enabled/disabled. So this makes sense, I'll make the change.

Reply:
Thread.sleep is not required anymore, however we need to pass the connection settings for refreshing the explorer if the region and credentials change during the time taken to enable/disable ecs exec